File: api/schemas/auth_result/auth_result.py
import graphene

from schemas.User.user import User


class AuthResult(graphene.ObjectType):
    """
    A type used to return information when users sign up or authenticate
    """
    auth_token = graphene.String(
        description="JWT used for accessing, controlled content."
    )
    user = graphene.Field(
        lambda: User,
        description="User that has just been created or signed in"
    )

File: api-js/src/types/auth-result.js
const { GraphQLObjectType, GraphQLString } = require('graphql')
const { userPersonalType } = require('./base')

const authResultType = new GraphQLObjectType({
  name: 'AuthResult',
  description: `An object used to return information when users sign up or authenticate.`,
  fields: () => ({
    authToken: {
      type: GraphQLString,
      description: `JWT used for accessing controlled content.`,
      resolve: ({ token }) => token,
    },
    user: {
      type: userPersonalType,
      description: `User that has just been created or signed in.`,
      resolve: ({ user }) => user,
    },
  }),
})

module.exports = {
  authResultType,
}
